### PR TITLE
Change data type to float[] and shape to int[] in JTensor

### PR DESCRIPTION
**Pull request by chengshunzhang**
Change the type of data in JTensor from List<Float> to float[], type of shape from List<Integer> to int[] and the corresponding parts in other files.

### CODE REVIEW

File: zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/AbstractInferenceModel.java [hunk 1]
```diff
@@ -71,16 +71,14 @@ public void reload(String modelPath, String weightPath) {
   }
 
   @Deprecated
-  public List<Float> predict(List<Float> input, int... shape) {
+  public List<Float> predict(float[] input, int... shape) {
```
Comment (yangw1234) [inline zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/AbstractInferenceModel.java]: do change existing api

File: zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/JTensor.java [hunk 2]
```diff
@@ -5,35 +5,62 @@
 import java.util.List;
 
 public class JTensor {
-  private List<Float> data;
+  private float[] data;
   private List<Integer> shape;
```
Comment (yangw1234) [inline zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/JTensor.java]: using int[] for shape?

File: zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/JTensor.java [hunk 3]
```diff
@@ -5,35 +5,62 @@
...
+    for (int i = 0; i < data.size(); i++){
+      this.data[i] = data.get(i);
+    }
+    this.shape = new ArrayList<Integer>();
+    for(int s : shape) {
+      this.shape.add(s);
+    }
+  }
+
+  public JTensor(float[] data, List<Integer> shape) {
```
Comment (yangw1234) [inline zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/JTensor.java]: copy data by default and provide an option for the no copy version

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/FloatInferenceModel.scala [hunk 4]
```diff
@@ -37,7 +37,7 @@ class FloatInferenceModel(
   @transient var predictor: LocalPredictor[Float]) extends InferenceSupportive with Serializable {
 
   @deprecated
-  def predict(input: JList[JFloat], shape: JList[JInt]): JList[JFloat] = {
```
Comment (yangw1234) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/FloatInferenceModel.scala]: do not change existing api

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceSupportive.scala [hunk 5]
```diff
@@ -74,11 +74,11 @@ trait InferenceSupportive {
...
 
   @inline
-  def transferInputToSample(input: JList[JFloat], inputShape: JList[JInt])
+  def transferInputToSample(input: Array[Float], inputShape: JList[JInt])
     : Sample[Float] = {
-    require(input.size() == product(inputShape), "data size not fit shape")
-    val inputData = toFloatArray(input)
-    Sample(Tensor(data = inputData, shape = toIntArray(inputShape)))
+    //require(input.size() == product(inputShape), "data size not fit shape")
+    //val inputData = toFloatArray(input)
```
Comment (yangw1234) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceSupportive.scala]: remove these two lines

File: zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/AbstractInferenceModel.java [hunk 6]
```diff
@@ -71,16 +71,14 @@ public void reload(String modelPath, String weightPath) {
...
     for (int s : shape) {
       inputShape.add(s);
     }
     return model.predict(input, inputShape);
   }
 
-  public List<List<JTensor>> predict(List<JTensor> inputs) {
-    return model.predict(inputs);
-  }
+  public List<List<JTensor>> predict(List<JTensor> inputs) { return model.predict(inputs); }
```
Comment (yangw1234) [inline zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/AbstractInferenceModel.java]: please change back for code style consistency

File: zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/AbstractInferenceModel.java (hunk 1 above)
Comment (chengshunzhang) [reply zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/AbstractInferenceModel.java]: But I will get an error 
Error:(79, 26) java: incompatible types: java.util.List<java.lang.Float> cannot be converted to float[]

Comment (chengshunzhang) [reply zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/AbstractInferenceModel.java]: I know what you mean now.

File: zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/JTensor.java (hunk 3 above)
Comment (chengshunzhang) [reply zoo/src/main/java/com/intel/analytics/zoo/pipeline/inference/JTensor.java]: Sorry, could you please be more explicit about this?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceSupportive.scala [hunk 7]
```diff
@@ -74,11 +74,9 @@ trait InferenceSupportive {
   }
 
   @inline
-  def transferInputToSample(input: JList[JFloat], inputShape: JList[JInt])
+  def transferInputToSample(input: Array[Float], inputShape: JList[JInt])
```
[the remark this answers is not in the document]
Comment (chengshunzhang) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceSupportive.scala]: OK

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceSupportive.scala [hunk 8]
```diff
@@ -74,11 +74,9 @@ trait InferenceSupportive {
   }
 
   @inline
-  def transferInputToSample(input: JList[JFloat], inputShape: JList[JInt])
-    : Sample[Float] = {
-    require(input.size() == product(inputShape), "data size not fit shape")
-    val inputData = toFloatArray(input)
-    Sample(Tensor(data = inputData, shape = toIntArray(inputShape)))
+  def transferInputToSample(input: Array[Float], inputShape: Array[Int])
```
Comment (yangw1234) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceSupportive.scala]: do we really need this function

**Comment by chengshunzhang** [comment]
Besides, in intellIJ, File -> Project structure -> Modules -> Sources, change the Language level to 8

**Pull request merged**